File: frontend/apps/website/src/typings/response.ts
import { JSONSchema } from '@/components/APIEditor/types'

export declare interface APICatCommonResponse {
  code: number
  id?: number | string
  name?: string
  description?: string
  content?: Record<string, { schema: JSONSchema }>
  $ref?: string
}

export declare interface APICatCommonResponseCustom {
  id: number | string
  expand: boolean
  isLocal: boolean
  isLoading: boolean
  isLoaded: boolean
  code?: number
  description?: string
  name?: string
  detail?: APICatCommonResponse
}

export declare interface DefinitionResponse {
  id?: number
  type: string
  name: string
  description?: string
  header?: Array<{
    name: string
    description?: string
    required?: boolean
    schema: JSONSchema
  }>
  content?: Record<string, { schema: JSONSchema }>
}

File: commom/apicat_struct/response.go
package apicat_struct

import "github.com/apicat/apicat/commom/spec"

type ResponseObjectList struct {
	List []*ResponseObject `json:"list"`
}

type ResponseObject struct {
	Name        string                 `json:"name"`
	Code        int                    `json:"code"`
	Description string                 `json:"description"`
	Header      []*Header              `json:"header,omitempty"`
	Content     map[string]spec.Schema `json:"content,omitempty"`
	Ref         string                 `json:"$ref,omitempty"`
}

type Header struct {
	Name        string      `json:"name"`
	Description string      `json:"description"`
	Example     string      `json:"example"`
	Default     string      `json:"default"`
	Required    bool        `json:"required"`
	Schema      spec.Schema `json:"schema"`
}
